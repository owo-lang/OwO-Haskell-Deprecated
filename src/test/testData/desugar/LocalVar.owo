module LocalVar where

 id = \x -> x
 flip = \f -> \x -> \y -> f y x
