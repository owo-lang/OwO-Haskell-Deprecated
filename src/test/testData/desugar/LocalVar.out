File type: owo
Top module name: .LocalVar
 implementation of function flip (38 4 2) (42 4 6)
 type
  meta variable _ (38 4 2) (42 4 6)
 value
  name binding
  it's a lambda binding
  bound scope
   name binding
   it's a lambda binding
   bound scope
    name binding
    it's a lambda binding
    bound scope
     function application
     function being applied
      function application
      function being applied
       local reference f (63 4 27) (64 4 28)
       referring to a binding f (46 4 10) (47 4 11)
      value applied to the function
       local reference y (65 4 29) (66 4 30)
       referring to a binding y (58 4 22) (59 4 23)
     value applied to the function
      local reference x (67 4 31) (68 4 32)
      referring to a binding x (52 4 16) (53 4 17)
 implementation of function id (24 3 2) (26 3 4)
 type
  meta variable _ (24 3 2) (26 3 4)
 value
  name binding
  it's a lambda binding
  bound scope
   local reference x (35 3 13) (36 3 14)
   referring to a binding x (30 3 8) (31 3 9)
